packages/astro/components/Markdown.astro

```diff
@@ -20,7 +20,7 @@ const { privateRenderMarkdownDoNotUse: renderMarkdown } = (Astro as any);
 if (!content) {
   const { privateRenderSlotDoNotUse: renderSlot } = (Astro as any);
   content = await renderSlot('default');
-  if (content.trim().length > 0) {
+  if (content !== undefined && content !== null) {
```

Comment:
This makes sense to me. Obviously we’d need to test this, but I can see the logic in just assuming strings are correct, and handling non-strings like you’re doing here.